lib/methods/ifwapichecktask.cpp

```diff
@@ -0,0 +1,460 @@
+					if (kv.second.GetType() == ValueObject) {
+						auto now (Utility::GetTime());
+
+						ReportIfwCheckResult(
+							checkable, command->GetName(), cr,
+							"$ifw_api_arguments$ may not directly contain objects (especially functions).", now, now
+						);
+
+						return;
+					}
+				}
+			}
+
+			/* MacroProcessor::ResolveArguments() converts
+			 *
+			 * [ "check_example" ]
+			 * and
+			 * {
+			 * 	 "-f" = { set_if = "$example_flag$" }
+			 * 	 "-a" = "$example_arg$"
+			 * }
+			 *
+			 * to
+			 *
+			 * [ "check_example", "-f", "-a", "X" ]
+			 *
+			 * but we need the args one-by-one like [ "-f" ] or [ "-a", "X" ].
+			 */
+			Array::Ptr arg = MacroProcessor::ResolveArguments(
+				emptyCmd, new Dictionary({{kv.first, argSpec}}), resolvers, lcr, resolvedMacros, useResolvedMacros
+			);
+
+			switch (arg ? arg->GetLength() : 0) {
+				case 0:
+					break;
+				case 1: // [ "-f" ]
+					params->Set(arg->Get(0), true);
+					break;
+				case 2: // [ "-a", "X" ]
+					params->Set(arg->Get(0), arg->Get(1));
+					break;
+				default: { // [ "-a", "X", "Y" ]
+					auto k (arg->Get(0));
+
+					arg->Remove(0);
+					params->Set(k, arg);
+				}
+			}
+		}
+	}
+
+	auto checkTimeout (command->GetTimeout());
+	auto checkableTimeout (checkable->GetCheckTimeout());
+
+	if (!checkableTimeout.IsEmpty())
+		checkTimeout = checkableTimeout;
+
+	if (resolvedMacros && !useResolvedMacros)
+		return;
+
+	if (psHost.IsEmpty()) {
+		psHost = "localhost";
+	}
+
+	if (expectedSan.IsEmpty()) {
+		expectedSan = IcingaApplication::GetInstance()->GetNodeName();
+	}
+
+	if (cert.IsEmpty()) {
+		cert = ApiListener::GetDefaultCertPath();
+	}
+
+	if (key.IsEmpty()) {
+		key = ApiListener::GetDefaultKeyPath();
+	}
+
+	if (ca.IsEmpty()) {
+		ca = ApiListener::GetDefaultCaPath();
+	}
+
+	Url::Ptr uri = new Url();
+
+	uri->SetPath({ "v1", "checker" });
+	uri->SetQuery({{ "command", psCommand }});
+
+	auto relative (uri->Format());
+	auto body (JsonEncode(params));
+	auto req (Shared<request<string_body>>::Make());
+
+	req->method(verb::post);
+	req->target(relative);
+	req->set(field::content_type, "application/json");
+	req->body() = body;
+
+	static auto curlTlsMinVersion ((String("--") + DEFAULT_TLS_PROTOCOLMIN).ToLower());
+
+	Array::Ptr cmdLine = new Array({
+		"curl", "--silent", "--show-error", curlTlsMinVersion, "--fail",
+		"--resolve", expectedSan + ":" + psPort + ":" + psHost,
+		"--ciphers", DEFAULT_TLS_CIPHERS,
+		"--cert", cert,
+		"--key", key,
+		"--cacert", ca,
+		"--request", "POST",
+		"--url", "https://" + expectedSan + ":" + psPort + relative,
+		"--header", "Content-Type: application/json",
+		"--data-raw", body
+	});
+
+	if (!crl.IsEmpty()) {
+		cmdLine->Add("--crlfile");
+		cmdLine->Add(crl);
+	}
+
+	if (!username.IsEmpty() && !password.IsEmpty()) {
+		auto authn (username + ":" + password);
+
+		req->set(field::authorization, "Basic " + Base64::Encode(authn));
+		cmdLine->Add("--user");
+		cmdLine->Add(authn);
+	}
```

Comment:
> I'm still not convinced that we want to render a fake curl command line here. We aren't using curl or even libcurl internally, so it probably won't match the behavior exactly (given that you even added TLS versions and ciphers, it looks to me like you were trying to achieve this).

I didn’t _try_ anything. I _achieved_ it. So IMAO this is not a fake, but an emulation. (OK, maybe ex. timeout, but shouldn't matter.)

> Bonus question regarding the usefulness of this: are you aware that in PowerShell, `curl` is an alias for `Invoke-WebRequest` by default?

Yes. And I've explicitly ignored it. curl is _curl_. End of story.

> Regarding the `--user` parameter: I wouldn't add the password to the command line and therefore to the check result. Yes, many check plugins take password as command line arguments (which is a bad idea in the first place) and therefore often passwords appear in the check results, but I wouldn't use that as an excuse here. For check plugins, you could avoid this by carefully choosing your plugins so that they also take passwords from environment variables

I -in contrast- would also render env vars for regular plugins for reproducible debugging. But that's another construction area.